examples/notebooks/get-started/learn_beam_transforms_by_doing.ipynb

```diff
@@ -0,0 +1,750 @@
+        "        | 'Find words' >> beam.FlatMap(lambda line: re.findall(r\"[a-zA-Z']+\", line))\n",
+        "        | 'Count words' >> beam.combiners.Count.PerElement()\n",
+        "        | 'Write results' >> beam.io.WriteToText(outputs_prefix)\n",
+        "  )\n",
+        "\n",
+        "pipeline.run()\n",
+        "# Sample the first 20 results, remember there are no ordering guarantees.\n",
+        "run('head -n 20 {}-00000-of-*'.format(outputs_prefix))"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "cellView": "form",
+        "id": "6Cn4pk3j8K4p"
+      },
+      "outputs": [],
+      "source": [
+        "#@title Answer\n",
+        "inputs_pattern = 'data/*'\n",
+        "outputs_prefix = 'outputs/part3'\n",
+        "\n",
+        "# Running locally in the DirectRunner.\n",
+        "with beam.Pipeline() as pipeline:\n",
+        "  word_count = (\n",
+        "      pipeline\n",
+        "        | 'Read lines' >> beam.io.ReadFromText(inputs_pattern)\n",
+        "        # The composite transform function you created above\n",
+        "        | 'Count Words' >> CountWords()\n",
+        "        | 'Write results' >> beam.io.WriteToText(outputs_prefix)\n",
+        "  )\n",
+        "\n",
+        "# Sample the first 20 results, remember there are no ordering guarantees.\n",
+        "run('head -n 20 {}-00000-of-*'.format(outputs_prefix))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "w73aU2IsDAS1"
+      },
+      "source": [
+        "### 3.3 Creating Your Own Composite Transform With `PTransform` Directly"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "uLEAICZKDz6e"
+      },
+      "source": [
+        "To create your own composite transform, create a subclass of the `PTransform` class and override the `expand` method to specify the actual processing logic\n",
+        "([more details](https://beam.apache.org/documentation/programming-guide/#composite-transform-creation))."
+      ]
+    },
+    {
+      "attachments": {},
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "JT_ZtzjIEHhD"
+      },
+      "source": [
+        "For example, if we wanted to create our own composite transform that counted the length of each word.\n",
+        "\n",
+        "The following code sample shows how to declare a `PTransform` that accepts a `PCollection` of Strings for input, and outputs a `PCollection` of integers\n",
+        "to show the string lengths.\n",
+        "\n",
+        "Within your `PTransform` subclass, you’ll need to override the `expand` method. The `expand` method is where you add the processing logic for the `PTransform`. \n",
+        "Your override of `expand` must accept the appropriate type of input `PCollection` as a parameter, and specify the output `PCollection` as the return value.\n",
```

Comment:
Maybe each of these paragraphs can have their own brief code cell as an example?

Reply:
This part is basically the complete example (check the full code below as well) for https://beam.apache.org/documentation/programming-guide/#composite-transform-creation. The only paragraph I think might need the example is this transform can contain other transforms but I feel like it would be better to keep the current code simple. Do you have better ideas with other examples?

Reply:
This is fine. As a follow up macro-AI, we should embed this into our documentation

Reply:
Yes, that is the next step.